tests/unit/amo/helpers.js

```diff
@@ -213,16 +214,34 @@ export const fakeRecommendations = Object.freeze({
   outcome: 'recommended_fallback',
 });
 
+export const onLocationChanged = ({ pathname, search = '', hash = '' }) => {
```

Comment:
Copying and pasting this seems a bit fragile but I see that you can't get it any other way.

I had an idea to use higher level APIs to mimic the behavior here but I'm not sure that it's any better. Here's a patch that works but I'm not sure it's a good idea. Anyway, just food for thought.

```diff
diff --git a/tests/unit/amo/helpers.js b/tests/unit/amo/helpers.js
index ea493cf88..ef45aef31 100644
--- a/tests/unit/amo/helpers.js
+++ b/tests/unit/amo/helpers.js
@@ -1,3 +1,4 @@
+import { createMemoryHistory } from 'history';
 import { normalize } from 'normalizr';
 import config from 'config';
 import { LOCATION_CHANGE } from 'connected-react-router';
@@ -240,7 +241,21 @@ export function dispatchClientMetadata({
   store.dispatch(setUserAgent(userAgent));
 
   // Simulate the behavior of `connected-react-router`.
-  store.dispatch(onLocationChanged({ pathname }));
+  const history = createMemoryHistory();
+  const unlisten = history.listen((location, action) => {
+    store.dispatch({
+      type: LOCATION_CHANGE,
+      payload: {
+        location,
+        action,
+      },
+    });
+  });
+  try {
+    history.push({ pathname });
+  } finally {
+    unlisten();
+  }
 
   return {
     store,
```

Reply:
Here's an idea: maybe just compare this `history.listen` approach to our copy/pasted action in a unit test to make sure they dispatch the same action. That would at least give us a failure in case our copy/pasted code gets out of sync.

Reply:
The problem I see here is that the `LOCATION_CHANGE` action is still hardcoded.

Reply:
I'll open a follow-up PR with a solution to this action being hardcoded here and copy/pasted in our helper file. I have not found a solution yet, but I am working on it.

Reply:
Filed: https://github.com/mozilla/addons-frontend/issues/6560

Reply:
I'm not so worried about hard coding the action type since it's imported from a constant. I'm more worried about the action payload going out of sync. The issue you filed looks good, thanks.